src/inject/dynamic-theme/stylesheet-proxy.ts

```diff
@@ -0,0 +1,46 @@
+export function injectProxy() {
+    if (!window['__darkreader__addRuleFunction']) {
+        window['__darkreader__addRuleFunction'] = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'addRule');
```

Comment:
You can just store the descriptors in the function scope like `const addRuleDescriptor = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'addRule')` :)